DESIGN.md

```diff
@@ -0,0 +1,47 @@
+# HTTP Provider Design
+
+The HTTP Provider offers functionality for interacting with generic HTTP servers as part of terraform configuration.
+Specifically, the provider issues an HTTP GET request to the defined URL on every Terraform run.
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide managed data source to issue HTTP GET request. The underlying default
+[transport](https://pkg.go.dev/net/http#Transport) uses [HTTP/1.1](https://datatracker.ietf.org/doc/html/rfc2616) for
+HTTP URLs and either [HTTP/1.1](https://datatracker.ietf.org/doc/html/rfc2616) or 
+[HTTP/2.0](https://datatracker.ietf.org/doc/html/rfc7540) for HTTPS URLs depending on whether the server supports
+[HTTP/2.0](https://datatracker.ietf.org/doc/html/rfc7540). Non-standard protocols (e.g., 
+[SPDY](https://tools.ietf.org/id/draft-ietf-httpbis-http2-00.html), 
+[QUIC](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-34)) are not supported.
+* Support usage of requests to URLs directed to either `http` or `https`. The current version of this provider is built
+with [Go 1.17](https://go.dev/doc/go1.17) which [supports](https://go.dev/doc/go1.17#minor_library_changes) 
```

Comment:
This is excellent! I wonder if it is worth calling out that Go updates may change this support over time based on the fluid security landscape? For example, I believe Go 1.18 removes TLS/1.0 and TLS/1.1 support. I don't think we need to call out the Go 1.18 thing explicitly here, but I think it is worth ensuring we are free to track Go upgrades according to the design.

Reply:
Have added a statement reflecting that TLS support will track the underlying Go version the provider is built with and will likely change over time.